site/src/components/PaginationWidget/PaginationNavButton.tsx

```diff
@@ -0,0 +1,84 @@
+import { type ReactNode, useEffect, useState } from "react";
+
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import { useTheme } from "@emotion/react";
+
+type PaginationNavButtonProps = {
+  children: ReactNode;
+  disabled: boolean;
+  disabledMessage: ReactNode;
+  onClick: () => void;
+  "aria-label": string;
+
+  disabledMessageTimeout?: number;
+};
+
+export function PaginationNavButton({
+  children,
+  onClick,
+  disabled,
+  disabledMessage,
+  "aria-label": ariaLabel,
+  disabledMessageTimeout = 3000,
+}: PaginationNavButtonProps) {
+  const theme = useTheme();
+  const [showDisabledMessage, setShowDisabledMessage] = useState(false);
+
+  // Inline state sync - this is safe/recommended by the React team in this case
+  if (!disabled && showDisabledMessage) {
+    setShowDisabledMessage(false);
+  }
+
+  useEffect(() => {
+    if (!showDisabledMessage) {
+      return;
+    }
+
+    const timeoutId = window.setTimeout(
```

Comment:
```suggestion
    const timeoutId = setTimeout(
```

Reply:
This doesn't make much of a difference in our code, but I have a habit of prefixing this with `window` to calls like this, so that TypeScript knows that we're dealing with the browser version of the APIs, instead of the Node.js versions (which can have slightly different type signatures)